{
  testConstraintMessageCodes(getConstraint("testInteger",new IntRange(1,5)),new Integer(7),new String[]{"testClass.testInteger.range.error","testClass.testInteger.size.toobig"},new Object[]{"testInteger",TestClass.class,new Integer(7),new Integer(1),new Integer(5)});
  testConstraintMessageCodes(getConstraint("testInteger",new IntRange(1,5)),new Integer(0),new String[]{"testClass.testInteger.range.error","testClass.testInteger.size.toosmall"},new Object[]{"testInteger",TestClass.class,new Integer(0),new Integer(1),new Integer(5)});
  testConstraintPassed(getConstraint("testString",new ObjectRange("abca","abcf")),"abcd");
  testConstraintPassed(getConstraint("testInteger",new IntRange(1,7)),new Integer(5));
  testConstraintPassed(getConstraint("testInteger",new IntRange(1,7)),null);
  testConstraintDefaultMessage(getConstraint("testInteger",new IntRange(1,5)),new Integer(7),"Property [{0}] of class [{1}] with value [{2}] does not fall within the valid range from [{3}] to [{4}]");
}
